### ISSUE TITLE
Dock does not scale itself correctly when moved to a monitor with different DPI

### ISSUE THREAD

**Issue by teoulas** [micheleg/dash-to-dock#435]
I'm using the master branch. I have 2 monitors: My primary one is a 3200x1800 one and Gnome scales everything to 2x there. My secondary one is a 1920x1200 and Gnome uses 1x scale there. However, if I move the dock there, it is still displayed at a 2x scale, instead of 1x.

To make this easier to grasp, I have attached 2 screenshots, with the dock and the settings window side-by-side. The settings window is scaled correctly, according to the DPI of the monitor. The dock is not.

Normal DPI monitor (secondary):
![96dpi](https://cloud.githubusercontent.com/assets/53618/22002829/aeb442fc-dc57-11e6-8d03-d7bb058d6965.jpg)

HiDPI monitor (primary):
![192dpi](https://cloud.githubusercontent.com/assets/53618/22002839/be004120-dc57-11e6-9fb0-643d6e30ee8b.jpg)

**Comment by micheleg** [comment]
I didn't know it were possible to have independent scale factors on different monitors. Is this under Wayland?

**Comment by teoulas** [comment]
Yes, it's only under Wayland. See here for more info: https://wiki.gnome.org/HowDoI/HiDpi

**Comment by micheleg** [comment]
Thanks for the link. To understand better the origin of the issue: is the default dash (extension disabled) shown with the right size (in X/Wayland), and on both monitors?

**Comment by teoulas** [comment]
To be honest, I don't know if that's possible. I can't seem to find a way to move the dash to another monitor without using your extension.

**Comment by micheleg** [comment]
You need to set the external monitor as primary in the display settings, so the whole overview moves there. That should work.

**Comment by teoulas** [comment]
Right, that worked! The dash is displayed correctly at 1x scale when the low DPI monitor is the primary. If I use dash to dock to move it to the high DPI monitor then it's still at 1x scale, instead of 2x.

**Comment by trickreich** [comment]
@micheleg Any news on this? Under Wayland with a mixed dpi setup (laptop + external monitor) this extension can sadly be used on the primary monitor.

**Comment by dblitt** [comment]
Hey! Is there any progress on the multi-dpi support? Btw thanks for the extension!

**Comment by yavko** [comment]
Yeah, anything on this?

**Comment by vanvugt** [comment]
Downstream Ubuntu bug: https://launchpad.net/bugs/1989503

**Comment by VGerris** [comment]
This still seems to be be in issue 6 years later, on Fedora 37 with Gnome 43 ? Any plans to fix this? Would be great, looks funny with double size icons ( 1 4k screen builtin, 2 external FHD screens, Wayland). Thank you

**Comment by proairface** [comment]
And almost a year later no comments in this thread... I will post one to "bump" and to raise awareness for this once again. Greetings to the person reading this in a (couple) year(s) time when coming across this finding that it still hasn't been resolved yet :/

**Comment by pustelnik89** [comment]
> And almost a year later no comments in this thread... I will post one to "bump" and to raise awareness for this once again. Greetings to the person reading this in a (couple) year(s) time when coming across this finding that it still hasn't been resolved yet :/

In my case enabling `scale-monitor-framebuffer` and rebooting resolved this issue: 
`gsettings set org.gnome.mutter experimental-features "['scale-monitor-framebuffer']"`
Maybe it will work for you too.

**Comment by proairface** [comment]
@pustelnik89 yes that worked for me, thanks!

**Comment by 3v1n0** [comment]
Scale monitors framebuffer is the way and it's default in gnome now.

Closing
